public_html/wp-content/mu-plugins/blocks/.eslintrc.js

```diff
@@ -137,6 +137,13 @@ module.exports = {
 			'destructuring': 'all',
 		} ],
 
+		/**
+		 * Disallow creating more than one component per file.
+		 *
+		 * Having one component per file makes components easier to test and easier to document.
+		 */
+		'react/no-multi-comp': [ 'error' ],
```

Comment:
Is this just for stateful components, or functional as well?

Reply:
This is for all components, but we could use `ignoreStateless` to allow functional components– as of right now I think even our functional components are complex enough that they could use the separate file + PropTypes approach, but I'm OK with allowing simple stateless components through.

Reply:
That would be my preference for a general rule, but I agree that there may be specific ones that warrant separating.

Things like `GridInspectorPanel` and `GridColumnsControl ` don't feel all that complex to me at first glance, but I'm open to it. I guess the thing I'm worried about is that having a ton of folders/files can be just as hard to navigate as having a component buried in a non-obvious file w/ other components, especially when everything is just named `index.js`.

Reply:
Ideally our folders would also have a better naming scheme where the folder name reflects the "top level" component (`index.js`), and each sibling component file name would match the component too - so `post-list/inspector-controls.js` would be `post-list/grid-inspector-panel.js` and `post-list/grid-columns-control.js`

But actually this one's not a great example, because I think `GridColumnsControl` could be merged into `GridInspectorPanel` directly since it's not used anywhere else.

Reply:
I mostly tried to leave file names the same so imports wouldn't change, but if we want to be more clear with files/folders I can also update the PRs to use that naming scheme too.

Reply:
I'm open to renaming stuff, but I agree it'd be more manageable if that were a separate issue.

Reply:
OK, once we've merged/decided on the sub-PRs here I can make another PR following that, but it doesn't make sense to start until this is settled.

Reply:
+1 for `ignoreStateless`